integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+
+        self.walter.set_pubkey_for_role(
+            "allow_all_transaction_signers",
+            "transactor.transaction_signer",
+            permit_keys=["*"],
+            deny_keys=[])
+
+        #
+        # transactor.transaction_signer.< tp_name > subsection
+        #
+        # From local configuration Dave is denied from being a batch_signer,
+        # Mallory and Chuck are denied being transactors.
+
+        self.walter.set_pubkey_for_role(
+            "deny_alice_from_xo_allow_bob",
+            "transactor.transaction_signer.xo",
+            permit_keys=[self.bob.public_key, self.dave.public_key],
+            deny_keys=[self.alice.public_key])
+
+        self.assert_able_to_send(self.bob, Families.XO)
+
+        for t, f in [(self.alice, Families.XO), (self.chuck, Families.XO),
+                     (self.mallory, Families.XO), (self.dave, Families.XO)]:
+            self.assert_not_able_to_send(t, f)
+
+        self.walter.set_pubkey_for_role(
+            "deny_bob_from_intkey_allow_dave_alice",
+            "transactor.transaction_signer.intkey",
+            permit_keys=[self.alice.public_key, self.dave.public_key],
+            deny_keys=[self.bob.public_key])
+
+        self.assert_able_to_send(self.alice, Families.INTKEY)
+
+        for t, f in [(self.bob, Families.INTKEY), (self.chuck, Families.INTKEY),
+                     (self.mallory, Families.INTKEY), (self.dave, Families.INTKEY)]:
+            self.assert_not_able_to_send(t, f)
+
+    def assert_able_to_send(self, transactor, family):
+        transactor.send(family)
+
+    def assert_able_to_send_batch(self, transactor, family, transactions):
+        transactor.send(family_name=family, transactions=transactions)
+
+    def assert_not_able_to_send(self, transactor, family):
+        with self.assertRaises(Exception):
+            transactor.send(family)
+
+    def assert_not_able_to_send_batch(self, transactor, family, txns):
+        with self.assertRaises(Exception):
+            transactor.send(family, txns)
+
+
+INTKEY_NAMESPACE = MessageFactory.sha512('intkey'.encode())[:6]
+XO_NAMESPACE = MessageFactory.sha512('xo'.encode())[:6]
+IDENTITY_NAMESPACE = '00001d'
+
+ROLE_PREFIX = '01'
+POLICY_PREFIX = '00'
+
+
+class Families(enum.Enum):
+    INTKEY = 1
+    XO = 2
+    IDENTITY = 3
+
+
+FamilyConfig = {
+    Families.INTKEY: {
+        'encoding': 'application/cbor',
+        'family_name': 'intkey',
+        'family_version': '1.0',
+        'namespace': MessageFactory.sha256('intkey'.encode())[:6]
+    },
+    Families.XO: {
+        'encoding': 'csv-utf8',
+        'family_name': 'xo',
+        'family_version': '1.0',
+        'namespace': MessageFactory.sha256('xo'.encode())[:6]
+    },
+    Families.IDENTITY: {
+        'encoding': 'application/protobuf',
+        'family_name': 'identity',
+        'family_version': '1.0',
+        'namespace': '00001d'
+    }
+}
+
+
+def INTKEY_PAYLOAD(unique_value):
```

Comment:
These are functions, not constants, so they should be named something like `make_intkey_payload`, etc.

Reply:
Updating.